### ISSUE TITLE
Cannot log in to GitHub actions

### ISSUE THREAD

**Issue by Zingo21** [github/vscode-github-actions#251]
**Describe the bug**
It is not possible to log in to Github actions. Even if I try to with the refresh button, it won't work.

**To Reproduce**
1. Open Github Actions extension
2. Click on log in
3. Afterwards it will show up in the extension that  it was `Unable to connect to the GitHub API, please check your internet connection.`
4. (Optional) See error in the output log when you click on refresh button in the extension `Cannot fetch github context
Error: Cannot fetch github context`

**Expected behavior**
I expected to be logged in to my GitHub account and se my actions in the repository I'm working on

**Screenshots**

From the output log
![Skärmavbild 2023-08-21 kl  16 27 05](https://github.com/github/vscode-github-actions/assets/82944718/468f05c2-b062-4683-85a9-c77e7cc51efb)

From the extension
![Skärmavbild 2023-08-21 kl  16 26 48](https://github.com/github/vscode-github-actions/assets/82944718/8adb7026-5fa6-4798-867e-ee64d7d8f0b1)

**Extension Version**
`v0.26.1`

**Comment by sebastianbuechler** [comment]
Same here:
```
Cannot fetch github context
Error: Cannot fetch github context
```

**Comment by Zingo21** [comment]
It might be the git authenticaton configuration because I have also the problem to push workflow files in VS Code and in an external terminal.

**Comment by monument-bmann** [comment]
I see this issue when VSCode has a workspace open. 

If I open an individual project I have no issue. 

```
Version v0.26.1

# from Window output when clicking refresh
2023-08-23 12:39:14.055 [error] command 'github-actions.explorer.refresh' not found: Error: command 'github-actions.explorer.refresh' not found
    at b.k (vscode-file://vscode-app/.../AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1669:3069)
    at b.executeCommand (vscode-file://vscode-app/.../AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1669:2985)
    at async i.open (vscode-file://vscode-app/.../AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:616:40250)
    at async d.open (vscode-file://vscode-app/.../AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:616:41761)

# from Github Actions output
Activating GitHub Actions extension...
Error getting GitHub context
Not Found
HttpError: Not Found
	at /home/{user}/.vscode-server/extensions/github.vscode-github-actions-0.26.1/dist/extension-node.js:2:1366275
	at runMicrotasks (<anonymous>)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async /home/{user}/.vscode-server/extensions/github.vscode-github-actions-0.26.1/dist/extension-node.js:2:1444089
	at async Promise.all (index 3)
	at async /home/{user}/.vscode-server/extensions/github.vscode-github-actions-0.26.1/dist/extension-node.js:2:1444005
	at async /home/{user}/.vscode-server/extensions/github.vscode-github-actions-0.26.1/dist/extension-node.js:2:1443637
	at async vt (/home/{user}/.vscode-server/extensions/github.vscode-github-actions-0.26.1/dist/extension-node.js:2:1443574)
	at async Os (/home/{user}/.vscode-server/extensions/github.vscode-github-actions-0.26.1/dist/extension-node.js:2:1624463)
	at async w.n (/home/{user}/.vscode-server/bin/1ad8d514439d5077d2b0b7ee64d2ce82a9308e5a/out/vs/workbench/api/node/extensionHostProcess.js:92:7006)
	at async w.m (/home/{user}/.vscode-server/bin/1ad8d514439d5077d2b0b7ee64d2ce82a9308e5a/out/vs/workbench/api/node/extensionHostProcess.js:92:6969)
	at async w.l (/home/{user}/.vscode-server/bin/1ad8d514439d5077d2b0b7ee64d2ce82a9308e5a/out/vs/workbench/api/node/extensionHostProcess.js:92:6426)

**Comment by Zingo21** [comment]
The git authentication config was not the problem. I thought that the problem lied there because I've had problems pushing GitHub action workflow files from VS Code. When I finally set up the git auth config, I could push commits that had the github actions workflow files, and then I thought that the git auth config might be the solution to the Github actions extension too. But it was not. I'm greeted with no context in the output and could not connect to the Github API.

**Comment by NSimpragaVolur** [comment]
Same issue here:

`Activating GitHub Actions extension...
Error getting GitHub context
Not Found
HttpError: Not Found
	at c:\Users\<user>\.vscode\extensions\github.vscode-github-actions-0.26.1\dist\extension-node.js:2:1366275
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async c:\Users\<user>\.vscode\extensions\github.vscode-github-actions-0.26.1\dist\extension-node.js:2:1444089
	at async Promise.all (index 0)
	at async c:\Users\<user>\.vscode\extensions\github.vscode-github-actions-0.26.1\dist\extension-node.js:2:1444005
	at async c:\Users\<user>\.vscode\extensions\github.vscode-github-actions-0.26.1\dist\extension-node.js:2:1443637
	at async vt (c:\Users\<user>\.vscode\extensions\github.vscode-github-actions-0.26.1\dist\extension-node.js:2:1443574)
	at async Os (c:\Users\<user>\.vscode\extensions\github.vscode-github-actions-0.26.1\dist\extension-node.js:2:1624463)
	at async p.n (c:\Users\<user>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:6206)
	at async p.m (c:\Users\<user>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:6169)
	at async p.l (c:\Users\<user>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:102:5626)`

**Comment by incomingstick** [comment]
I seem to receive a similar, but not identical error in my GitHub Actions output:

```
Error getting GitHub context
fetch failed
HttpError: fetch failed
	at c:\Users\<user>\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1366802
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
	at async c:\Users\<user>\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1447997
	at async Object.get (c:\Users\<user>\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1447724)
	at async Lt (c:\Users\<user>\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1447916)
	at async c:\Users\<user>\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1626346
	at async n.h (c:\Users\<user>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:125:84275)

Cannot fetch github context
Error: Cannot fetch github context
	at vt (c:\Users\<user>\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1444927)
	at async c:\Users\<user>\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1626446
	at async n.h (c:\Users\<user>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:125:84275)
```

My while my GitHub is:
```
2023-10-13 10:29:49.462 [info] Log level: Info
2023-10-13 10:29:52.627 [warning] Failed to update repository branch protection: No GitHub authentication session available.
2023-10-13 10:29:59.528 [warning] Failed to update repository branch protection: No GitHub authentication session available.
2023-10-13 10:30:28.560 [warning] Failed to update repository branch protection: No GitHub authentication session available.
```

The GitHub Pull Requests and Issues extensions appears to work without issue. It is only GitHub Actions that fails, the same way as the initial issue. This is for an Enterprise org as well, however no GitHub Enterprise URI is required.

When I click the "Refresh" button in the GitHub Actions extension, I can see it use the GitHub Authentication extension successfully get the session:

```
2023-10-13 10:40:48.493 [info] Getting sessions for repo,workflow...
2023-10-13 10:40:48.493 [info] Got 1 sessions for repo,workflow...
```

**Comment by sgordon46** [comment]
I'm receiving a similar experience and similar output in the logs. 

```
2023-10-26 11:33:46.034 [info] Getting sessions for repo,workflow...
2023-10-26 11:33:46.034 [info] Got 1 sessions for repo,workflow...
```

```Cannot fetch github context
Error: Cannot fetch github context
	at vt (/Users/<user>/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1444927)
	at async /Users/<user>/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1626446
	at async n.h (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:125:84275)
```

These are the messages I receive when my corporate VPN is running, but if I disable the VPN all is working fine.

Access to github.com is working fine, along with my other vscode GitHub integrations, both with VPN and without.

**Comment by ArminPrieschl** [comment]
Same issue here with Github Enterprise and SAML Auth.

```
Cannot fetch github context
Error: Cannot fetch github context
    at vt (c:\Users\user\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1444927)
    at Bt (c:\Users\user\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1445257)
    at Os (c:\Users\user\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1637593)
    at v.n (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:125:6205)
    at v.m (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:125:6168)
    at v.l (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:125:5625)
...initialized
```

**Comment by moongoal** [comment]
Experiencing the same issue on Windows, extension v0.26.2, vscode latest.

**Comment by bng5** [comment]
Probably not the same issue. In my case, when clicking on "Sign in to GitHub" this error is printed in the Developer Tools console:
```
  ERR command 'github-actions.sign-in' not found: Error: command 'github-actions.sign-in' not found
    at y.n (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:1537:12331)
    at y.executeCommand (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:1537:12260)
    at async i.open (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:645:41218)
    at async u.open (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:645:42600)
```

  * OS: Ubuntu 20.04.6 LTS
  * GitHub Actions v0.26.2
  * VS Code
    * 1.84.2
    * 1a5daa3a0231a0fbba4f14db7ec463cf99d7768e
    * x64

**Comment by TomokiOshiman** [comment]
Same here
* OS: M1 Mac
* GitHub Actions: v0.26.2

```
Cannot fetch github context
Error: Cannot fetch github context
    at vt (/Users/user/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1444927)
    at /Users/user/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1626446
    at d.h (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:141:140086)
```

**Comment by renatho** [comment]
Same here on M1, extension v0.26.2.

Already tried opening a repository folder or opening VS code without any project selected. Both presented the same issues.

The first output is:
```
Activating GitHub Actions extension...
Error getting GitHub context
Bad credentials
HttpError: Bad credentials
```

And every time I click on "Refresh" for the extension, I receive the following output:

```
Cannot fetch github context
```

**Comment by alexgshaw** [comment]
I also have this same issue:

```
Activating GitHub Actions extension...
Error getting GitHub context
Bad credentials
HttpError: Bad credentials
    at /Users/alexshaw/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1366400
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at /Users/alexshaw/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1447997
    at Object.get (/Users/alexshaw/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1447724)
    at Lt (/Users/alexshaw/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1447916)
    at Os (/Users/alexshaw/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1624572)
    at u.n (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:6255)
    at u.m (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:6218)
    at u.l (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:5675)

Cannot fetch github context
Error: Cannot fetch github context
    at vt (/Users/alexshaw/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1444927)
    at Os (/Users/alexshaw/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1624588)
    at u.n (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:6255)
    at u.m (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:6218)
    at u.l (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:5675)
```

I have a Macbook with an Intel chip. Other GitHub extensions work fine, e.g. GitHub Copilot

**Comment by saqib-ahmed** [comment]
Same issue for me on Macbook Pro intel with Sonoma Version 14.2.1 (23C71).

```
Activating GitHub Actions extension...
Error getting GitHub context
Not Found
HttpError: Not Found
	at /Users/xyz/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1366400
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
	at async /Users/xyz/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1444214
	at async Promise.all (index 0)
	at async /Users/xyz/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1444130
	at async /Users/xyz/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1443762
	at async vt (/Users/xyz/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1443699)
	at async Os (/Users/xyz/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1624588)
	at async u.n (/Users/xyz/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:6255)
	at async u.m (/Users/xyz/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:6218)
	at async u.l (/Users/xyz/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:5675)
```

**Comment by gzimbron** [comment]
Same issue. 

Macbook Pro M1 Pro, actually I have Github copilot working on my vscode, but error in Github Actions extension

```
Cannot fetch github context
Error: Cannot fetch github context
    at vt (/Users/gzimbron/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1444927)
    at /Users/gzimbron/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1626446
    at s.h (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:150:175235)
    at /Users/gzimbron/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1625230
```

**Comment by GingricB** [comment]
I have found a solve to this issue.

There is an extension in VS code called "win-ca" that is used to map local certifications into VS code including GitHub. This is for windows users only. For mac users, use "Mac CA VSCode". I was able to set the config of that extension to append certs (defaults to replace) and it now allows me to fetch the GitHub API when cloning a repository and using the GitHub Actions extension in VS code.

My issue occurred only when I was on our companies VPN but works fine now with the extension I listed above.

Windows: https://marketplace.visualstudio.com/items?itemName=ukoloff.win-ca
MAC: https://marketplace.visualstudio.com/items?itemName=linhmtran168.mac-ca-vscode

**Comment by bryant-finney** [comment]
I started experiencing this issue after I upgraded to `1.87.0-insider` ([`8fa84589`](https://github.com/microsoft/vscode/commits/8fa84589eef3538dbc763ff98dc7d5a8a0c56374)) a couple of days ago. It had been working just fine for months before the upgrade.

The issue is present on both my M2 Mac (Sonoma `14.2.1`) and my Windows 11 Pro (version `23H2`). As others describe, only the [GitHub.vscode-github-actions](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-github-actions) extension is affected; I'm successfully using all of the following:
- [GitHub.copilot](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot)
- [GitHub.copilot-chat](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-chat)
- [GitHub.remotehub](https://marketplace.visualstudio.com/items?itemName=GitHub.remotehub)
- [GitHub.vscode-pull-request-github](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github)

I'm seeing the same traceback as others have shared:

```js
Error: Cannot fetch github context
    at vt (/home/bryantfinney/.vscode-server-insiders/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1444927)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at /home/bryantfinney/.vscode-server-insiders/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1626446
    at s.h (/home/bryantfinney/.vscode-server-insiders/bin/8fa84589eef3538dbc763ff98dc7d5a8a0c56374/out/vs/workbench/api/node/extensionHostProcess.js:150:178999
```

- OS: 
  - Windows 11 Pro `23H2` (both on the host and on a `WSL2` distro of `Ubuntu 22.04`)
  - M2 Mac (Sonoma `14.2.1`)
- VS Code:
  - `1.87.0-insider`
  - [`8fa84589eef3538dbc763ff98dc7d5a8a0c56374`](https://github.com/microsoft/vscode/commits/8fa84589eef3538dbc763ff98dc7d5a8a0c56374)
  - x64 (windows) | arm64 (macos)

**Comment by scottco-slalom** [comment]
> I have found a solve to this issue.
> 
> There is an extension in VS code called "win-ca" that is used to map local certifications into VS code including GitHub. This is for windows users only. For mac users, use "Mac CA VSCode". I was able to set the config of that extension to append certs (defaults to replace) and it now allows me to fetch the GitHub API when cloning a repository and using the GitHub Actions extension in VS code.
> 
> My issue occurred only when I was on our companies VPN but works fine now with the extension I listed above.
> 
> Windows: https://marketplace.visualstudio.com/items?itemName=ukoloff.win-ca MAC: https://marketplace.visualstudio.com/items?itemName=linhmtran168.mac-ca-vscode

This is what finally solved the issue for me.  Thank you so much for sharing the extension.  Literally all I did was install Mac CA VSCode and restart, and the GitHub Actions extension starting working as epected for the first time in ages!

**Comment by mysticflute** [comment]
Experiencing this issue as well on a Mac. Tried uninstalling and reinstalling but it didn't work.

**Comment by doudar** [comment]
Same issue here. The win-ca extension didn't help me.

**Comment by AaronMT** [comment]
`Mac CA VSCode` did nothing for me either.

```Activating GitHub Actions extension...
Error getting GitHub context
Bad credentials
HttpError: Bad credentials
	at /Users/aaronmt/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1366400
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
	at async /Users/aaronmt/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1447997
	at async Object.get (/Users/aaronmt/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1447724)
	at async Lt (/Users/aaronmt/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1447916)
	at async Os (/Users/aaronmt/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1624572)
	at async u.n (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:6255)
	at async u.m (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:6218)
	at async u.l (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:5675)
```

and then on every attempt to refresh

```
Cannot fetch github context
Error: Cannot fetch github context
    at vt (/Users/aaronmt/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1444927)
    at /Users/aaronmt/.vscode/extensions/github.vscode-github-actions-0.26.2/dist/extension-node.js:2:1626446
    at l.h (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:150:184831)
```

**Comment by KamasamaK** [comment]
Using VS Code 1.87.2 on Windows 10

```
Cannot fetch github context
Error: Cannot fetch github context
	at vt (c:\Users\kama\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1444927)
	at async c:\Users\kama\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1626446
	at async l.h (c:\Users\kama\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:150:184831)
Error getting GitHub context
Bad credentials
HttpError: Bad credentials
	at c:\Users\kama\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1366400
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
	at async c:\Users\kama\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1447997
	at async Object.get (c:\Users\kama\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1447724)
	at async Lt (c:\Users\kama\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1447916)
	at async hs.refresh (c:\Users\kama\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1614307)
	at async c:\Users\kama\.vscode\extensions\github.vscode-github-actions-0.26.2\dist\extension-node.js:2:1626804
	at async l.h (c:\Users\kama\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:150:184831)
```

I can confirm that the GitHub credentials are good as the "GitHub Pull Requests" extension works fine with the same GitHub Authentication account.

**Comment by EagleIJoe** [comment]
Version 0.26.2 fixed the SAML login issue for me. After a vscode restart it asked me to reauth because SAML enforcement is enabled, and that did the trick.

**Comment by MetRonnie** [comment]
Enabling Experimental System Certificates v2 (in settings) did not fix it for me. This is still the only GitHub extension that doesn't auth for me

**Comment by cortexcompiler** [comment]
I think this may be a duplicate of: https://github.com/github/vscode-github-actions/issues/39.
At least, I know my issue is that it is not using the certificates in the keystore and therefore getting blocked by a corporate proxy.

**Comment by mechaniccoder** [comment]
I resolve this issue by steps below.

1. Sign out from github by right click account menu in vscode (left bottom in sidebar)
![Screenshot 2024-05-25 at 8 54 32 PM](https://github.com/github/vscode-github-actions/assets/58871413/2e6d24ce-77a3-4860-9071-de4b25124d35)

2. Reauthenticate github by clicking button in github actions extension

**Comment by garysassano** [comment]
How can you work on a GitHub workflow from an organization's private repo  when you are logged in your personal account in VS Code to sync all your settings and extensions?

**Comment by pratheesh-prakash** [comment]
> I resolve this issue by steps below.
> 
>     1. Sign out from github by right click account menu in vscode (left bottom in sidebar)
>        ![Screenshot 2024-05-25 at 8 54 32 PM](https://private-user-images.githubusercontent.com/58871413/333797715-2e6d24ce-77a3-4860-9071-de4b25124d35.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3ODA5MDUsIm5iZiI6MTcxODc4MDYwNSwicGF0aCI6Ii81ODg3MTQxMy8zMzM3OTc3MTUtMmU2ZDI0Y2UtNzdhMy00ODYwLTkwNzEtZGU0YjI1MTI0ZDM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE5VDA3MDMyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcwNzE3Y2ZjZjZiZGY4ZGU0OTFmODBkNWNiZjU0NTgwNmQwYjJmYTI3OGQ5OGYwOTg1Y2E4YTNjNmYyYjkyNWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7tcwVDB81rgNUBIyHX816P3Y4kzQ2OHu9CrhlUi2H6U)
> 
>     2. Reauthenticate github by clicking button in github actions extension

This worked for me.

**Comment by linyihai** [comment]
> I resolve this issue by steps below.
> 
> 1. Sign out from github by right click account menu in vscode (left bottom in sidebar)
>    ![Screenshot 2024-05-25 at 8 54 32 PM](https://private-user-images.githubusercontent.com/58871413/333797715-2e6d24ce-77a3-4860-9071-de4b25124d35.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3MDI3MjEsIm5iZiI6MTcyMTcwMjQyMSwicGF0aCI6Ii81ODg3MTQxMy8zMzM3OTc3MTUtMmU2ZDI0Y2UtNzdhMy00ODYwLTkwNzEtZGU0YjI1MTI0ZDM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIzVDAyNDAyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFiOWEwYjcyMTdiODE4OWFjMGJhZDMzMTQ3Yzg0Y2FhYmU1Yjk5ZGQ5MzEyMDI3MjkwZDExMjRkODgxZGE5MTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Xup1QgVnnQGa5ib25TFx20XynKyfumuBfKDWvuRqUrc)
> 2. Reauthenticate github by clicking button in github actions extension

This works for me. I recently changed my password, but the vscode inner github login state doesn't updated.

**Comment by mfleyfel** [comment]
Anyone still having this problem, this solved it for me in Windows after everything else here did nothing:
"Launch VS Code, go to File > Preferences > Settings > Search for "certificates" and check the box for Http > Experimental: System Certificates V2 "x Controls whether experimental loading of CA certificates from the OS should be enabled."

[https://stackoverflow.com/questions/36506539/how-do-i-get-visual-studio-code-to-trust-our-self-signed-proxy-certificate](https://stackoverflow.com/questions/36506539/how-do-i-get-visual-studio-code-to-trust-our-self-signed-proxy-certificate)

![fix](https://github.com/user-attachments/assets/dfd97f6e-8864-41b1-bbdc-4cf5be67f005)

**Comment by hejfelix** [comment]
> I have found a solve to this issue.
> 
> There is an extension in VS code called "win-ca" that is used to map local certifications into VS code including GitHub. This is for windows users only. For mac users, use "Mac CA VSCode". I was able to set the config of that extension to append certs (defaults to replace) and it now allows me to fetch the GitHub API when cloning a repository and using the GitHub Actions extension in VS code.
> 
> My issue occurred only when I was on our companies VPN but works fine now with the extension I listed above.
> 
> Windows: https://marketplace.visualstudio.com/items?itemName=ukoloff.win-ca MAC: https://marketplace.visualstudio.com/items?itemName=linhmtran168.mac-ca-vscode

this worked for me on mac os with an SSO login. Thanks!!

**Comment by gjsjohnmurray** [comment]
For me the solution was:
* Have one of my GH Actions YAML files open.
* Run `Accounts: Manage Trusted Extensions For Account` from Command Palette.
* Choose my GitHub account.
* Deselect the "GitHub Actions" checkbox and click OK.
* Run `Developer: Reload Window`
* On the "Sign in to GitHub to access your repositories and GitHub Actions workflows." notification click "Sign in to GitHub".
* Allow the GitHub Actions to sign in.

**Comment by aceslick911** [comment]
> * Run `Accounts: Manage Trusted Extensions For Account` from Command Palette.
> * Choose my GitHub account.
> * Deselect the "GitHub Actions" checkbox and click OK.
> * Run `Developer: Reload Window`
> * On the "Sign in to GitHub to access your repositories and GitHub Actions workflows." notification click "Sign in to GitHub".

Thankyou! Was banging my head all morning on this one! Definetely feels like this is a bug with the extension 👍

**Comment by Zingo21** [comment]
> For me the solution was:
> 
> * Have one of my GH Actions YAML files open.
> * Run `Accounts: Manage Trusted Extensions For Account` from Command Palette.
> * Choose my GitHub account.
> * Deselect the "GitHub Actions" checkbox and click OK.
> * Run `Developer: Reload Window`
> * On the "Sign in to GitHub to access your repositories and GitHub Actions workflows." notification click "Sign in to GitHub".
> * Allow the GitHub Actions to sign in, choosing my GH account on the quickpick.

I tried these steps, but I still got "Unable to connect to the GitHub API, please check your internet connection.

**Comment by thezanke** [comment]
I had this issue, to get it working here's what I did:
- Open extensions, right click the "GitHub Actions" extension and select "Account Preferences"
- Click "Use new account" even though my desired account is listed and selected already
- Sign in to the account like normal
- Right click the GitHub Actions extension again and disable it; restart extensions as prompted
- Right click the extension again and re-enable it
  - It will ask you to allow it to use your auth
- Select the desired account and then refresh the actions pane it should work

**Comment by MetRonnie** [comment]
If this is caused by a proxy:

> Make sure your proxy allows connections to all hosts/domains listed here: https://code.visualstudio.com/docs/setup/network
